crates/node/core/src/args/pruning.rs

```diff
@@ -3,48 +3,248 @@
+        } else {
+            None
+        }
+    }
+
+    const fn receipts_prune_mode(&self) -> Option<PruneMode> {
+        if self.receipts_full {
+            Some(PruneMode::Full)
+        } else if let Some(distance) = self.receipts_distance {
+            Some(PruneMode::Distance(distance))
+        } else if let Some(block_number) = self.receipts_before {
+            Some(PruneMode::Before(block_number))
+        } else {
+            None
+        }
+    }
+
+    const fn account_history_prune_mode(&self) -> Option<PruneMode> {
+        if self.account_history_full {
+            Some(PruneMode::Full)
+        } else if let Some(distance) = self.account_history_distance {
+            Some(PruneMode::Distance(distance))
+        } else if let Some(block_number) = self.account_history_before {
+            Some(PruneMode::Before(block_number))
+        } else {
+            None
+        }
+    }
+
+    const fn storage_history_prune_mode(&self) -> Option<PruneMode> {
+        if self.storage_history_full {
+            Some(PruneMode::Full)
+        } else if let Some(distance) = self.storage_history_distance {
+            Some(PruneMode::Distance(distance))
+        } else if let Some(block_number) = self.storage_history_before {
+            Some(PruneMode::Before(block_number))
+        } else {
+            None
+        }
+    }
+
+    fn parse_receipts_log_filter(&self) -> Result<ReceiptsLogPruneConfig, String> {
+        let mut config = BTreeMap::new();
+        // Each filter is provided in the form <address>:<prunemode>
+        for filter in &self.receipts_log_filter {
+            let parts: Vec<&str> = filter.split(':').collect();
+            if parts.len() < 2 {
+                return Err(format!("Invalid filter format: {}", filter));
+            }
+            let address =
+                parts[0].parse::<Address>().map_err(|e| format!("Invalid address: {}", e))?;
+            let prune_mode = match parts[1] {
+                "full" => PruneMode::Full,
+                s if s.starts_with("distance:") => {
+                    let distance = s
+                        .strip_prefix("distance:")
+                        .unwrap()
+                        .parse::<u64>()
+                        .map_err(|e| format!("Invalid block distance: {}", e))?;
+                    PruneMode::Distance(distance)
+                }
+                s if s.starts_with("before:") => {
+                    let block_number = s
+                        .strip_prefix("before:")
+                        .unwrap()
+                        .parse::<BlockNumber>()
+                        .map_err(|e| format!("Invalid block number: {}", e))?;
+                    PruneMode::Before(block_number)
+                }
+                _ => return Err(format!("Invalid prune mode: {}", parts[1])),
+            };
+            config.insert(address, prune_mode);
+        }
+        Ok(ReceiptsLogPruneConfig(config))
```

Comment:
one or two unit tests for this would be good. at least one with well formed input, possibly also one with malformed input.